user_guide_src/source/testing/overview.rst

```diff
@@ -256,14 +258,31 @@ component name:
 
 .. note:: All component Factories are reset by default between each test. Modify your test case's ``$setUpMethods`` if you need instances to persist.
 
+.. _testing-overview-stream-filters:
+
 Stream Filters
 ==============
 
-**CITestStreamFilter** provides an alternate to these helper methods.
+**StreamFilterTrait** provides an alternate to these helper methods.
 
 You may need to test things that are difficult to test. Sometimes, capturing a stream, like PHP's own STDOUT, or STDERR,
-might be helpful. The ``CITestStreamFilter`` helps you capture the output from the stream of your choice.
+might be helpful. The ``StreamFilterTrait`` helps you capture the output from the stream of your choice.
+
+**Overview of methods**
+``StreamFilterTrait::getStreamFilterBuffer()`` Get the captured data from the buffer.
+``StreamFilterTrait::resetStreamFilterBuffer()`` Reset captured data.
 
 An example demonstrating this inside one of your test cases:
 
 .. literalinclude:: overview/018.php
+
+The ``StreamFilterTrait`` has a configurator that is called automatically.
+See :ref:`testing-overview-traits`.
+
+If you override the ``setUp()`` or ``tearDown()`` methods in your test, then you must call the ``parent::setUp()`` and
+``parent::tearDown()`` methods respectively to configure the ``StreamFilterTrait``.
+
+If your test does not inherit from the ``CIUnitTestCase`` class, then you need to call the ``setUpStreamFilterTrait()``
+and ``tearDownStreamFilterTrait()`` methods.
+
+.. literalinclude:: overview/019.php
```

Comment:
This sample code is difficult to understand, because it is mixture of overriding setUp and not extending CIUnitTestCase.